process/block/preprocess/transactions.go

```diff
@@ -1186,8 +1210,10 @@ func (txs *transactions) computeSortedTxs(
 	log.Debug("computeSortedTxs.GetSortedTransactions")
 	sortedTxs := sortedTransactionsProvider.GetSortedTransactions()
 
-	SortTransactionsBySenderAndNonce(sortedTxs)
-	return sortedTxs, nil
+	selectedTxs := txs.preFilterTransactions(sortedTxs, gasBandwidth)
```

Comment:
the preFilter function should go inside the sortedTransactionsProvider - You should push the gasBandwidth there

Reply:
prefiltering is only required when one is creating the block, so for the consensus group leader, not for observers or the other validators. 
All validators/observers are calling the sortTransactions, this is why it was not moved inside.

Reply:
No. GetSortedTransactions is called only by the leader on the CreateBlock flow. It is not called by validators, obsservers.

Reply:
I somehow thought of the sorting that is done afterwards. 
Yes this could theoretically be moved there but in this case the logic needs to be changed more. 
I will leave this as a todo.